[aotinductor] Fix a missing schema issue for repeat_interleave

desertfire opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #110267
* #110233
* __->__ #110105



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

Differential Revision: [D49686812](https://our.internmc.facebook.com/intern/diff/D49686812)

desertfire committed:
[aotinductor] Fix a missaing schema issue for repeat_interleave

[ghstack-poisoned]

desertfire committed:
Update on "[aotinductor] Fix a missaing schema issue for repeat_interleave"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

desertfire committed:
Update on "[aotinductor] Fix a missaing schema issue for repeat_interleave"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

desertfire committed:
Update on "[aotinductor] Fix a missaing schema issue for repeat_interleave"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

desertfire committed:
Update on "[aotinductor] Fix a missaing schema issue for repeat_interleave"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

desertfire committed:
Update on "[aotinductor] Fix a missing schema issue for repeat_interleave"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

desertfire commented:
@desertfire has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D49686812).

desertfire committed:
Update on "[aotinductor] Fix a missing schema issue for repeat_interleave"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov

Differential Revision: [D49686812](https://our.internmc.facebook.com/intern/diff/D49686812)

[ghstack-poisoned]

zou3519 commented on torch/_inductor/ir.py:
```diff
@@ -3630,21 +3630,42 @@ def __init__(
-            if isinstance(kernel, torch._ops.OpOverload)
-            else kernel
-        )
+        # TODO: Need to revisit schema matching to find the correct OpOverload from OpOverloadPacket
```
For the future, I don't think inductor should ever get an OpOverloadPacket and we should maintain that invariant. AOTAutograd should normalize things into OpOverload/HigherOrderOperator (and we've defined "ATen IR" to only consist of OpOverload/HigherOrderOperator), so is the only source of OpOverloadPacket additional passes after AOTAutograd and before we get here?

desertfire replied:
It's because in _inductor/lowering.py, we do something like https://github.com/pytorch/pytorch/blob/7c1702f099f4301a3f124fc7bf697f3234d7f9d2/torch/_inductor/lowering.py#L1860 which passes `OpOverloadPacket` to `make_fallback`. We can turn that into `OpOverload` with `.default`, but that probably requires more cleaning in cpp codegen. I will separate into another PR.

zou3519 commented on torch/_inductor/ir.py:
```diff
@@ -3630,21 +3630,42 @@ def __init__(
+                    # repeat_interleave(const at::Tensor & self, int64_t repeats,
+                    #       c10::optional<int64_t> dim=c10::nullopt, c10::optional<int64_t> output_size=c10::nullopt)
+                    #
+                    # In theory, we should be able to do the following codegen for all aten fallback ops, but more
+                    # plumbing needs to be done in order to make that happen.
+                    # TODO: follow up on this
```
Do you mean you want to do this for OpOverloadPacket in addition to OpOverload, or also OpOverload that are the "default" overload?

If it is the latter, you can just move the kernel._overloadname thing into the true branch: 
```
kernel_name = kernel.__name__.replace('.', '_') if kernel._overloadname != "default" else kernel.__name__.replace('.default', '')
self.kernel = f"at::_ops::{kernel_name}::call"
```

desertfire replied:
No, I meant putting `schema = kernel._schema` into the else branch will cause some other tests to fail, which I will fix as a follow up.

desertfire committed:
Update on "[aotinductor] Fix a missing schema issue for repeat_interleave"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov

Differential Revision: [D49686812](https://our.internmc.facebook.com/intern/diff/D49686812)

[ghstack-poisoned]

desertfire commented:
@pytorchbot merge

desertfire commented:
@desertfire has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D49686812).

desertfire commented:
@pytorchbot merge
